### PR TITLE
Improved documentation for SafeMath class

### PR DESCRIPTION
**Pull request by windushka**
#129

### CODE REVIEW

File: packages/as-types/assembly/safeMath.ts [hunk 1]
```diff
@@ -1,8 +1,18 @@
+/**
+ * The SafeMath class provides utility functions for basic arithmetic operations.
+ * These functions perform overflow and underflow checks to prevent unwanted behavior
+ * when dealing with unsigned 64-bit integers (u64). The SafeMath class should be used
+ * when working with arithmetic operations that require increased safety and precision.
+ */
 export class SafeMath {
   /**
+   * add - Returns the addition of two unsigned 64-bit integers (u64),
```
Comment (0xSwapFeeder) [inline packages/as-types/assembly/safeMath.ts:9]: 'function_name - title' Is this some kind of standard way to make function titles in TSDoc ?
@Ben-Rey @gregLibert what do you think ? Should we apply this rule every where in documentation ?

Comment (Ben-Rey) [reply packages/as-types/assembly/safeMath.ts:9]: No need to repeat the name of the function in the title.
The title is just a description of what the function does
https://tsdoc.org/

**Review by 0xSwapFeeder — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/safeMath.ts [hunk 2]
```diff
@@ -1,8 +1,29 @@
...
+ * @remarks
+ * The SafeMath class is designed to be a drop-in replacement for standard arithmetic
+ * operations on unsigned 64-bit integers. By using the methods provided by this class,
+ * developers can avoid potential overflow and underflow issues and ensure that their
+ * code behaves correctly even in edge cases.
+ */
 export class SafeMath {
   /**
+   * Returns the addition of two unsigned 64-bit integers (u64),
+   *       reverting on overflow.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/safeMath.ts:16]:
```suggestion
   * Safely adds two unsigned 64-bit integers (u64), reverting if an
   * overflow occurs.
```

File: packages/as-types/assembly/safeMath.ts [hunk 3]
```diff
@@ -12,9 +33,18 @@ export class SafeMath {
   }
 
   /**
+   * Returns the subtraction of two unsigned 64-bit integers (u64),
+   *       reverting on underflow.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/safeMath.ts:37]:
```suggestion
   * Safely subtracts two unsigned 64-bit integers (u64), reverting if an
   * underflow occurs.
```

File: packages/as-types/assembly/safeMath.ts [hunk 4]
```diff
@@ -24,9 +54,18 @@ export class SafeMath {
   }
 
   /**
+   * Returns the multiplication of two unsigned 64-bit integers (u64),
+   *       reverting on overflow.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/safeMath.ts:58]:
```suggestion
   * Safely multiplies two unsigned 64-bit integers (u64), reverting on
   * overflow.
```

File: packages/as-types/assembly/safeMath.ts [hunk 5]
```diff
@@ -40,9 +79,18 @@ export class SafeMath {
   }
 
   /**
+   * Returns the integer division of two unsigned 64-bit integers (u64),
+   *       Reverts on division by zero. The result is rounded towards zero
```
Comment (Ben-Rey) [inline packages/as-types/assembly/safeMath.ts:83]:
```suggestion
   * Safely divides two unsigned 64-bit integers (u64), reverting on
   * division by zero. The result is rounded towards zero.
```

**Review by Ben-Rey — CHANGES_REQUESTED** [review]

**Review by 0xSwapFeeder — APPROVED** [review]
Lgtm :+1

**Review by Ben-Rey — APPROVED** [review]

File: packages/as-types/assembly/safeMath.ts [hunk 6]
```diff
@@ -1,8 +1,29 @@
...
-   * reverting on overflow.
+   * @remarks
+   * This function is used to safely add two unsigned 64-bit integers without causing an overflow.
+   *
+   * @param a - The first operand for addition.
+   * @param b - The second operand for addition.
+   *
+   * @returns The sum of a and b as an unsigned 64-bit integer (u64).
+   *
+   * @throws Throws an error if the operation results in an overflow.
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:26]: To be consistent with what [we did for as-sdk](https://github.com/massalabs/massa-as-sdk/blob/main/assembly/std/storage.ts#L170-L173):
```suggestion
   * @throws if the operation results in a number bigger than u64.MAX_VALUE.
```

File: packages/as-types/assembly/safeMath.ts [hunk 7]
```diff
@@ -12,9 +33,18 @@ export class SafeMath {
...
-   * division by zero. The result is rounded towards zero.
+   * @remarks
+   * This function is used to safely substract two unsigned 64-bit integers without causing an underflow.
+   *
+   * @param a - The first operand for subtraction (minuend).
+   * @param b - The second operand for subtraction (subtrahend).
+   *
+   * @returns The difference between a and b as an unsigned 64-bit integer (u64).
+   *
+   * @throws Throws an error if the operation results in an underflow.
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:47]:
```suggestion
   * @throws if the operation results in a number lower than 0.
```

File: packages/as-types/assembly/safeMath.ts [hunk 8]
```diff
@@ -24,9 +54,18 @@ export class SafeMath {
...
+   *
+   * @remarks
+   * This function is used to safely mutilply two unsigned 64-bit integers without causing an overflow.
+   *
+   * @param a - The first operand for multiplication.
+   * @param b - The second operand for multiplication.
+   *
+   * @returns The product of a and b as an unsigned 64-bit integer (u64).
+   *
+   * @throws Throws an error if the operation results in an overflow.
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:68]:
```suggestion
   * @throws if the operation results in a number bigger than u64.MAX_VALUE.
```

File: packages/as-types/assembly/safeMath.ts [hunk 9]
```diff
@@ -40,9 +79,18 @@ export class SafeMath {
   }
 
   /**
-   *
-   * @returns Returns the integer division of two unsigned integers. Reverts on
+   * Safely divides two unsigned 64-bit integers (u64), reverting on
    * division by zero. The result is rounded towards zero.
+   *
+   * @remarks
+   * This function is used to safely divide two unsigned 64-bit integers without causing a division by zero error.
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:86]: You should add a remarks that nothing is done to check lose in precision, the rest is ignored.
For instance, what happens if you do 4/3

File: packages/as-types/assembly/safeMath.ts [hunk 10]
```diff
@@ -40,9 +79,18 @@ export class SafeMath {
...
+   *
+   * @param a - The dividend,
+   * @param b - The divisor,
+   *
+   * @returns The quotient of a divided by b as an unsigned 64-bit integer (u64).
+   *
+   * @throws Throws an error if the operation results in a division by zero.
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:93]:
```suggestion
   * @throws if the operation results in a division by zero.
```

File: packages/as-types/assembly/safeMath.ts [hunk 11]
```diff
@@ -40,9 +79,18 @@ export class SafeMath {
...
   /**
-   *
-   * @returns Returns the integer division of two unsigned integers. Reverts on
+   * Safely divides two unsigned 64-bit integers (u64), reverting on
    * division by zero. The result is rounded towards zero.
+   *
+   * @remarks
+   * This function is used to safely divide two unsigned 64-bit integers without causing a division by zero error.
+   *
+   * @param a - The dividend,
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:88]:
```suggestion
   * @param a - The dividend.
```

File: packages/as-types/assembly/safeMath.ts [hunk 12]
```diff
@@ -40,9 +79,18 @@ export class SafeMath {
...
+   * @param b - The divisor,
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:89]:
```suggestion
   * @param b - The divisor.
```

File: packages/as-types/assembly/safeMath.ts [hunk 13]
```diff
@@ -40,9 +79,18 @@ export class SafeMath {
...
+   *
+   * @returns The quotient of a divided by b as an unsigned 64-bit integer (u64).
```
Comment (gregLibert) [inline packages/as-types/assembly/safeMath.ts:91]:
```suggestion
   * @returns The quotient of `a` divided by `b` as an unsigned 64-bit integer (u64).
```

**Review by gregLibert — CHANGES_REQUESTED** [review]

**Pull request merged**